File: 0791 custom-sort-string/solution.go
func customSortString(order string, s string) string {
    fr:=make([]int,26)
    for _,v:=range s {
        fr[v-'a']++
    }
    ans:=""
    for _,v := range order {
        for fr[v-'a'] > 0 {
            ans+=string(v)
            fr[v-'a']--
        }
    }
    for _,v := range s {
        if fr[v-'a'] > 0 {
            ans+=string(v)
            fr[v-'a']--
        }
    }
    return ans
}

File: 0791 custom-sort-string/solution.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        vector<int> fr(26,0);
        for(char c:s) fr[c-'a']++;
        string ans = "";
        for(char c:order) {
            while(fr[c-'a'] > 0) {
                ans.push_back(c);
                fr[c-'a']--;
            }
        }
        for(char c:s) {
            if(fr[c-'a'] > 0) {
                ans.push_back(c);
                fr[c-'a']--;
            }
        }
        return ans;
    }
};
